src/Json.Schema.ToDotNet/ComparerCodeGenerator.cs

```diff
@@ -0,0 +1,1603 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.Json.Schema.ToDotNet
+{
+    /// <summary>
+    /// Generates classes that implement <see cref="System.Collections.Generic.IComparer{T}"/>.
+    /// </summary>
+    internal class ComparerCodeGenerator
+    {
+        internal const string InstancePropertyName = "Instance";
+
+        private const string UriClassName = "Uri";
+        private const string GenericTypeName = "T";
+        private const string ObjectTypeName = "object";
+        private const string CountPropertyName = "Count";
+        private const string GenericKeyTypeName = "TKey";
```

Comment:
>GenericKeyTypeName [](http://example.com/codeflow?start=29&length=18)

Aren't there a lot of shared constants here between this code generator and the one for IEqualityComparer?

Reply:
The answer is no.